diff -Nur bt_workspace-org/qcom-opensource/bt/bt-app/rsp/src/Rsp.cpp bt_workspace/qcom-opensource/bt/bt-app/rsp/src/Rsp.cpp
--- bt_workspace-org/qcom-opensource/bt/bt-app/rsp/src/Rsp.cpp	2019-10-29 15:46:40.182536496 +0800
+++ bt_workspace/qcom-opensource/bt/bt-app/rsp/src/Rsp.cpp	2019-10-29 15:48:48.810541397 +0800
@@ -621,12 +621,14 @@
     fprintf(stdout," set rsp data \n");
     SetRSPAttrData(&rev);
     RegisterApp();
+    return true;
 }
 
 bool Rsp::DisableRSP()
 {
     fprintf(stdout, "(%s) Disable RSP Initiated",__FUNCTION__);
     StopService();
+    return true;
 }
 
 bool Rsp::RegisterApp()
@@ -681,6 +683,7 @@
                                                 IncludeName, IncludeTxPower, min_conn_interval,
                                                 max_conn_interval, 0,strlen(str), str,
                                                 strlen(str), str, 0,NULL);
+    return true;
 }
 
 void Rsp::CleanUp(int server_if)
