diff --git a/drivers/base/firmware_class.c b/drivers/base/firmware_class.c
index 2f23c4b..df967c7 100755
--- a/drivers/base/firmware_class.c
+++ b/drivers/base/firmware_class.c
@@ -381,6 +381,7 @@ static int fw_get_filesystem_firmware(struct device *device,
 		if (IS_ERR(file))
 			continue;
 		rc = fw_read_file_contents(file, buf);
+                printk("**********[DEBUG]***********path: %s, rc=%d\n", path, rc);
 		fput(file);
 		if (rc)
 			dev_warn(device, "firmware, attempted to load %s, but failed with error %d\n",
